### PR TITLE
Update CICE and use EMC/CICE:develop branch from CICE-Consortium:main

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [x] Commit 'test_changes.list' from previous step
---
## Description:

This PR changes the EMC/CICE branch used in UFS. While EMC/CICE:emc/develop branch has effectively same code and answers as authoritative CICE-Consortium:main branch, development history over years has introduced git commit differences. Merging code histories as well facilitates future two-way code management. For this, EMC commits ahead of Consortium have been merged to Consortium. This also syncs recent commits at Consortium, like an improved physics option for congelation ice formation.

Issuing a new EMC/CICE:develop branch preserves rather than rewrites history. 

### Commit Message:

```
* UFSWM - Use EMC/CICE:develop branch from CICE-Consortium:main
  * CICE - Use EMC/CICE:develop branch from CICE-Consortium:main
```

### Priority:

* Normal

### UFSWM:

* Closes #https://github.com/NOAA-EMC/CICE/issues/84
* Closes #https://github.com/NOAA-EMC/CICE/issues/82

### Sub component Pull Requests:

* None

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* No Baseline Changes.

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by NickSzapiro-NOAA** [comment]
fyi @NeilBarton-NOAA @rgrumbine @DeniseWorthen , use of new `congel_freeze = 'one-step'` physics namelist option may be recommended (https://github.com/CICE-Consortium/Icepack/pull/494 ), if anyone wants to do physics runs. It is not default and must be switched on

File: .gitmodules [hunk 1]
```diff
@@ -28,8 +28,8 @@
   branch = dev/emc
 [submodule "CICE"]
   path = CICE-interface/CICE
-  url = https://github.com/NOAA-EMC/CICE
-  branch = emc/develop
+  url = https://github.com/CICE-Consortium/CICE
```
Comment (junwang-noaa) [inline .gitmodules]: Does this mean we will use the CICE consortium main branch directly in UFS WM? That means CICE consortium code managers will get involved in the EMC cice code updates. Can we still use EMC's cice fork with a new develop branch? e.g.

url = https://github.com/NOAA-EMC/CICE
  branch = develop

Comment (NickSzapiro-NOAA) [reply .gitmodules]: I only used Consortium branch for testing b4b. Yes, we will use a EMC/CICE develop branch in UFS

Comment (NickSzapiro-NOAA) [reply .gitmodules]: I just made https://github.com/NOAA-EMC/CICE/tree/develop from CICE-Consortium. I didn't realize I had permissions

**Comment by jkbk2004** [comment]
I will see if we can apply a fix for fire component submodule pointer to https://github.com/NOAA-EMC/fire_behavior/commit/05cad173feeb598431e3ef5f17c2df6562c8d101 @zach1221 @FernandoAndrade-NOAA @BrianCurtis-NOAA FYI

**Comment by jkbk2004** [comment]
https://github.com/NOAA-EMC/fire_behavior/commit/05cad173feeb598431e3ef5f17c2df6562c8d101 reproduces b2b: /scratch2/NCEPDEV/marine/Jong.Kim/UFS-RT/rt-2430/tests/RegressionTests_hera.log. @NickSzapiro-NOAA can you update .gitmodules?
```
[submodule "fire_behavior"]
  path = fire_behavior
  url = https://github.com/NOAA-EMC/fire_behavior
  branch = emc/develop
```

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA You are updating CICE, but I see no associated CICE PR.

@jkbk2004 You marked the PR as ready. The subcomponent PR (I think there should be one) is not reviewed.

This PR is not ready for the commit Q.

**Comment by jkbk2004** [comment]
@NickSzapiro-NOAA can you make sure this pr is to bring up https://github.com/NOAA-EMC/CICE/tree/develop? The develop branch is fully ready, right?

**Comment by DeniseWorthen** [comment]
@jkbk2004 This PR will switch to a new develop branch, which includes recent updates from the Consortium. Those updates need a PR.

**Comment by NickSzapiro-NOAA** [comment]
@DeniseWorthen I don't know how to make an EMC/CICE PR for this. Maybe I made a mistake by creating a https://github.com/NOAA-EMC/CICE/tree/develop branch already during review. I can make an isssue with what is updated but no code should change (via PR)

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Despite the history not aligning, our actual code was always up-to-date w/ Consortium when we made a PR to update the top of our fork. You are doing two things---one, you're renaming the top of our fork, but you're also updating to the the latest in Consortium. You mention that this includes, for example, 'improved physics for congelation ice'. We need updates like that documented in our fork, so we know what changed between the top of the old emc/develop and the top of the new develop.

**Comment by NickSzapiro-NOAA** [comment]
@DeniseWorthen I've now listed the updates in an updated EMC/CICE issue (https://github.com/NOAA-EMC/CICE/issues/82). Are you asking to delete the current EMC/CICE develop branch to create it via PR? My working idea is that creating a subcomponent branch/tag/release/... can be done at any point but using new branch in UFS requires testing and approval.

**Comment by BrianCurtis-NOAA** [comment]
I'm not sure how you wish to handle this, but the current setup passes all RT's on Acorn. If you update the hash to the new develop branch of NOAA-EMC/CICE then that would be the only difference, assuming the switch from emc/develop to develop doesn't change anything in the code.

**Comment by NickSzapiro-NOAA** [comment]
Thanks @BrianCurtis-NOAA. There are hash and code changes between current emc/develop and develop branches of EMC/CICE

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Thanks for creating the issue. Since the new develop branch is already in place, I guess we can continue from here. I think it might have been better to do it in two steps, but that is my fault for not paying more attention.

**Comment by BrianCurtis-NOAA** [comment]
@NickSzapiro-NOAA what are we still waiting on for this PR to start?

**Comment by NickSzapiro-NOAA** [comment]
Just for testing to complete. The EMC/CICE branch already exists and we use it via .gitmodules as is

**Comment by zach1221** [comment]
Testing is done. Please give final reviews.

**Review by jkbk2004 — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**